homeassistant/components/climate/zwave.py

```diff
@@ -213,29 +188,24 @@ def set_temperature(self, **kwargs):
         else:
             return
 
-        self.set_value(
-            class_id=zwave.const.COMMAND_CLASS_THERMOSTAT_SETPOINT,
-            index=self._index, data=temperature)
+        self.values.primary.data = temperature
         self.schedule_update_ha_state()
```

Comment:
Probably right. It should get updated on the value changed callback. I'd rather leave that separate though, since I don't have a climate device to test with.